src/ucp/tag/rndv.c

```diff
@@ -52,20 +62,85 @@ size_t ucp_tag_rndv_pack_rkey(ucp_request_t *sreq, ucp_lane_index_t lane,
 
     /* Check if the sender needs to register the send buffer -
      * is its datatype contiguous and does the receive side need it */
-    if (ucp_ep_rndv_md_flags(ep) & UCT_MD_FLAG_NEED_RKEY) {
+    if (ucp_ep_rndv_md_flags(ep, 0) & UCT_MD_FLAG_NEED_RKEY) {
         status = ucp_request_send_buffer_reg(sreq, lane);
         ucs_assert_always(status == UCS_OK);
 
         /* if the send buffer was registered, send the rkey */
         UCS_PROFILE_CALL(uct_md_mkey_pack, ucp_ep_md(ep, lane),
-                         sreq->send.state.dt.dt.contig.memh, rkey_buf);
+                         sreq->send.state.dt.dt.contig[0].memh, rkey_buf);
         *flags |= UCP_RNDV_RTS_FLAG_PACKED_RKEY;
         return ucp_ep_md_attr(ep, lane)->rkey_packed_size;
     }
 
     return 0;
 }
 
+static size_t ucp_tag_rndv_pack_rkeys(ucp_request_t *sreq, void *rkey_buf, uint16_t *flags)
+{
+    ucp_ep_t        *ep      = sreq->send.ep;
+    ucp_dt_state_t  *state   = &sreq->send.state.dt;
+    size_t           packet  = 0;
+    int              cnt     = 0;
+    int              i;
+    ucp_lane_index_t lane;
+
+    ucs_assert(UCP_DT_IS_CONTIG(sreq->send.datatype));
+
+    cnt = ucp_request_rndv_reg(sreq);
+    ucs_assert_always(cnt <= UCP_MAX_RNDV_LANES);
+
+    if (cnt) {
```

Comment:
not sure why this check is needed, the for loop below would work even if no rkeys